### PR TITLE
Interactive investigation guides enhancements in 8.9 

### LINKED ISSUE #3188: Interactive investigation guides enhancements in 8.9
Document the latest enhancements or changes to the interactive investigation guides feature (ongoing work done in 8.6 with https://github.com/elastic/security-docs/issues/2696 and in 8.7 with https://github.com/elastic/security-docs/issues/2955). 

Also, the interactive investigation guide feature is set to go GA in the future, and will probably have some licensing/subscription requirements. Exact release TBA.

Related:
- https://github.com/elastic/kibana/issues/153377
   From issue description:
    > Scope of this feature will now include 3 parts: 
    > 1. hardening of the rendered button, so that we optimistically use fields from old versions with sensible defaults, 
    > 2. ability to delete timeline notes so in the event that invalid markdown is saved as a note, it can be removed by users
    > 3. wrapping the markdown renderer in a React ErrorBoundary, to prevent any rendering exceptions and also display any relevant parsing errors to the user.

- Epic: https://github.com/elastic/security-team/issues/4824

## To Do

- [x] Add info on backwards compatibility, any other new developments
- [x] Update screenshot with new icon in editor
- [x] Remove "technical preview" badge from screenshots
- [x] Relocate page out of [Technical preview](https://www.elastic.co/guide/en/security/master/sec-experimental-intro.html) section

### PR DESCRIPTION
**Pull request by joepeeples**
Resolves #3188.

Preview: [Launch Timeline from investigation guides](https://security-docs_3503.docs-preview.app.elstc.co/guide/en/security/master/interactive-investigation-guides.html)

Related: The callout about version compatibility needed to be revised for all versions going back to 8.6. But because this PR makes many other changes that don't need to be backported -- including relocating the file `investigation-guide-actions.asciidoc` -- updateing to 8.6-8.8 will need to be handled a little differently:
- This PR handles updates for `main` and `8.9` (the latter via Mergify).
- https://github.com/elastic/security-docs/pull/3505 will update `8.8`, then I'll use the backport tool to backport the change to `8.7` and `8.6`.

### CODE REVIEW

File: docs/detections/investigation-guide-actions.asciidoc [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [[interactive-investigation-guides]]
-== Interactive investigation guides
+== Launch Timeline from investigation guides
 
-IMPORTANT: This feature is available for {stack} versions 8.6.0 and newer. Additionally, interactive investigation guides are not compatible between release versions, so query buttons created in 8.6.0 won't render correctly in {version} and vice versa.
+IMPORTANT: Interactive investigation guides are compatible between {stack} versions 8.7.0 and later. Query buttons created in 8.6.x use different syntax and won't render correctly in later versions, and vice versa.
```
Comment (benironside) [inline docs/detections/investigation-guide-actions.asciidoc:4]: Perhaps the IMPORTANT callout could go after the paragraph following it so readers have context first.

**Review by benironside — COMMENTED** [review]
Looks good to me! Left just one idea for you in the comments.

File: docs/detections/investigation-guide-actions.asciidoc (hunk 1 above)
Comment (joepeeples) [reply docs/detections/investigation-guide-actions.asciidoc:4]: Sounds good to me, thanks!

**Review by nastasha-solomon — APPROVED** [review]
LGTM! 🚢🚢🚢

**Review by kqualters-elastic — APPROVED** [review]

**Pull request merged**